db/db_impl/db_impl_write.cc

```diff
@@ -1801,7 +1804,24 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
   }
   if (s.ok()) {
     SequenceNumber seq = versions_->LastSequence();
-    new_mem = cfd->ConstructNewMemtable(mutable_cf_options, seq);
+    std::unique_ptr<uint32_t> new_filterbits;
+
+    if (mutable_cf_options.memtable_self_tuning_bloom) {
+      const uint64_t pastentries = cfd->mem()->BFUniqueEntryEstimate();
+      new_filterbits.reset(new uint32_t);
+      // Calculate new BF size by assuming the number of unique keys
+      // will be similar to the mutable memtable being switched,
+      // (for 6 probes (hash functions) and a target FPR of 1%)
+      // m = - n_approx * ln(0.01)/[ln(2)**2]
+      *new_filterbits = static_cast<uint32_t>(
+          (pastentries * (-1.0) * (-2.0 * M_LN10 /* ln(0.01) */)) /
+          (M_LN2 * M_LN2));
+      TEST_SYNC_POINT_CALLBACK("DBImpl::SelfTuningBloom:NewBFSize",
+                               new_filterbits.get());
+    }
```

Comment:
I rememebr seeing an upper limit of 25% memtable bloom ratio here: https://github.com/facebook/rocksdb/blob/c625b8d017d43d0bceffbd7c0cf039708578fd1f/db/column_family.cc#L260 
Shoudl we also maybe update that value? Unless the user might want to reduce the FP rate down from 1%, I think based on https://github.com/facebook/rocksdb/pull/8671#discussion_r691696982 we may want to update this threshold.

Reply:
Actually the rose case scenario you be a key of size 2 bytes and a value fo size 1 byte, because then the encoded form  stored in the memtable is 1 + (2+8) + 1 + 1 = 13 bytes, details here: 
https://github.com/facebook/rocksdb/blob/c625b8d017d43d0bceffbd7c0cf039708578fd1f/db/db_flush_test.cc#L354
So maybe the upper hard limit should be 10% memtable Bloom Ratio. I dont see why a ratio > 10% would be necessary beside lowering the FPR

Reply:
I was including skiplist overhead of 1.33 * 8 bytes per entry. I believe only the vector memtable would have significantly less space overhead, and that should be rarely used (though bloom filter would pay off a lot for misses!).

Reply:
But 10% seems reasonable for self-tuning firewall limit if you want to use that.